backend/src/test/java/com/phaete/backend/forage/service/ForageMapItemServiceTest.java

```diff
@@ -0,0 +1,335 @@
+package com.phaete.backend.forage.service;
+
+import com.phaete.backend.forage.model.*;
+import com.phaete.backend.forage.repository.ForageMapItemRepository;
+import org.junit.jupiter.api.Test;
+
+import java.util.List;
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.*;
+
+class ForageMapItemServiceTest {
+
+	private final ForageMapItemRepository forageMapItemRepository = mock(ForageMapItemRepository.class);
+	private final IdService idService = mock(IdService.class);
+
+	@Test
+	void createForageMapItem_expectDTO_onSuccess() {
+		ForageMapItemDTO expectedForageMapItemDTO = new ForageMapItemDTO(
+				new ForageWikiItem(
+						"1",
+						"Apple Tree",
+						ForageCategory.FRUIT,
+						ForageSource.TREE,
+						"Apple Tree",
+						ForageSeason.FALL,
+						List.of("test")
+				),
+				new CustomMarker(
+						"1",
+						new double[] {0.0, 0.0},
+						"",
+						new int[] {0, 0},
+						new int[] {0, 0},
+						new int[] {0, 0},
+						""
+				),
+				ForageQuantity.ABUNDANT,
+				ForageQuality.EXCELLENT,
+				"never",
+				"notes"
+		);
+
+		when(forageMapItemRepository.save(any(ForageMapItem.class))).thenReturn(
+				new ForageMapItem(
+						"1",
+						new ForageWikiItem(
+								"1",
+								"Apple Tree",
+								ForageCategory.FRUIT,
+								ForageSource.TREE,
+								"Apple Tree",
+								ForageSeason.FALL,
+								List.of("test")
+						),
+						new CustomMarker(
+								"1",
+								new double[] {0.0, 0.0},
+								"",
+								new int[] {0, 0},
+								new int[] {0, 0},
+								new int[] {0, 0},
+								""
+						),
+						ForageQuantity.ABUNDANT,
+						ForageQuality.EXCELLENT,
+						"never",
+						"notes"
+				)
+		);
+		when(idService.generateId()).thenReturn("1");
+
+		ForageMapItemService forageMapItemService = new ForageMapItemService(
+				forageMapItemRepository,
+				new ConverterService(idService)
+		);
```

Comment:
Why do you create new Service objects in each test instead of once for the class?

Reply:
That's how I learned it. There's not really any downside afaik.

Reply:
You have declared the IdService as global variable. Why not the ConverterService?

And when the ForageMapItemService is also global, you have four detracting lines less in the test.